low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -31,19 +41,169 @@ class ServerMap(ResourceMap):
+            phys = self.sysfs.get_phy_list_for_port(port)
+            # eg: [ 'phy-1:0', 'phy-1:1', 'phy-1:2', 'phy-1:3']
+            specifis = {
+                'num_phys': len(phys),
+                'phys': []
+            }
+            health = "OK"
+            for phy in phys:
+                phy_data = self.sysfs.get_sas_phy_data(phy)
+                specifis['phys'].append(phy_data)
+                if phy_data['state'] != 'enabled' or \
+                   'Gbit' not in phy_data['negotiated_linkrate']:
+                    health = "Failed"
+
+            port_data['health']['specifics'].append(specifis)
+            self.set_health_data(port_data, health)
+            sas_ports_data.append(port_data)
+        return sas_ports_data
+
+    def get_nw_ports_info(self):
+        """Return the Network ports information."""
+        network_cable_data = []
+        io_counters = psutil.net_io_counters(pernic=True)
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            nic_info = self.get_data_template(interface, False)
+
+            specifics = {}
+            for addr in addrs:
+                if addr.family == socket.AF_INET:
+                    specifics["ipV4"] = addr.address
+
+            if interface in io_counters:
+                io_info = io_counters[interface]
+                specifics.update({
+                    "networkErrors": io_info.errin + io_info.errout,
+                    "droppedPacketsIn": io_info.dropin,
+                    "droppedPacketsOut": io_info.dropout,
+                    "packetsIn": io_info.packets_recv,
+                    "packetsOut": io_info.packets_sent,
+                    "trafficIn": io_info.bytes_recv,
+                    "trafficOut": io_info.bytes_sent
+                })
+
+            nw_status, nw_cable_conn_status = \
+                self.get_nw_status(interface)
+
+            specifics["nwStatus"] = nw_status
+            specifics["nwCableConnStatus"] = nw_cable_conn_status
+            nic_info["health"]["specifics"].append(specifics)
+
+            # set health status, description and recommendation
+            map_status = {"UP": "OK", "DOWN": "Failed", "UNKNOWN": "UNKNOWN"}
+            health_status = map_status[nw_cable_conn_status]
+
+            self.set_health_data(nic_info, health_status)
+
+            network_cable_data.append(nic_info)
+
+        return network_cable_data
+
+    def get_nw_status(self, interface):
+        """Read & Return the latest network status from sysfs files."""
+        try:
+            nw_status = self.sysfs.fetch_nw_operstate(interface)
+            nw_cable_conn_status = self.sysfs.fetch_nw_cable_status(
+                self.sysfs.get_sys_dir_path('net'), interface
+            )
+        except Exception as err:
+            raise ResourceMapError(errno.EINVAL,
+                                   ("Unable to fetch network status due to"
+                                    " an error: %s" % err))
```

Comment:
DO NOT raise exception anywhere.
Raising Exception should only happen at rpath validation.
Otherwise the thread will be closed and data collection with sub resources will be lost.